File: poet-spring-boot-starter/src/main/java/com/skypyb/poet/spring/boot/core/client/route/Navigation.java
package com.skypyb.poet.spring.boot.core.client.route;

import java.io.Serializable;
import java.util.Objects;

public class Navigation implements Serializable {

    private String name;

    private String module;

    private String fullPath;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getModule() {
        return module;
    }

    public void setModule(String module) {
        this.module = module;
    }

    public String getFullPath() {
        return fullPath;
    }

    public void setFullPath(String fullPath) {
        this.fullPath = fullPath;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Navigation that = (Navigation) o;
        return Objects.equals(name, that.name) &&
                Objects.equals(module, that.module) &&
                Objects.equals(fullPath, that.fullPath);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, module, fullPath);
    }
}

File: poet-spring-boot-starter/src/main/java/com/skypyb/poet/spring/boot/core/model/Navigation.kt
package com.skypyb.poet.spring.boot.core.model

import java.io.Serializable

data class Navigation(
        var name: String,

        var module: String,

        //path == key
        var path: String,
        /**
         * fullPath 和 path 不同的是： fullPath包含文件系统基础路径,即存储路径
         * 如果是unix系统, 则最前面带个斜杠 ‘/’
         */
        var fullPath: String
) : Serializable